app/workers/twitter_checkin_share_worker.rb

```diff
@@ -2,106 +2,88 @@ class TwitterCheckinShareWorker
   include Sidekiq::Worker
 
   def perform(checkin_id)
-    @rest = 0
     @checkin = Checkin.find(checkin_id)
 
     if @checkin.twitter_url_hash.blank?
       @checkin.update_column(:twitter_url_hash, @checkin.generate_url_hash)
     end
 
-    twitter = @checkin.user.providers.where(name: 'twitter').first
+    twitter = @checkin.user.providers.find_by(name: "twitter")
     client = Twitter::REST::Client.new do |config|
-      config.consumer_key        = ENV['TWITTER_CONSUMER_KEY']
-      config.consumer_secret     = ENV['TWITTER_CONSUMER_SECRET']
-      config.access_token        = twitter.token
+      config.consumer_key = ENV["TWITTER_CONSUMER_KEY"]
+      config.consumer_secret = ENV["TWITTER_CONSUMER_SECRET"]
+      config.access_token = twitter.token
       config.access_token_secret = twitter.token_secret
     end
 
-    client.update(tweet_text(@checkin.comment))
+    client.update(tweet_body(@checkin))
   end
 
 
   private
 
-  def get_work_title
-    max_length = 35
-    title = @checkin.work.title
-
-    if @checkin.comment.present?
-      if max_length < title.length
-        title.truncate(max_length)
-      else
-        @rest += (max_length - title.length)
-        title
-      end
-    else
-      title.truncate(max_length)
-    end
+  def tweet_body(checkin)
+    data = {
+      work_title: checkin.work.title,
+      episode_number: get_episode_number(checkin),
+      episode_title: get_episode_title(checkin),
+      share_url: "#{ENV['ANNICT_URL']}/r/tw/#{checkin.twitter_url_hash}",
+      share_hashtag: get_share_hashtag(checkin),
+      comment: checkin.comment
+    }
+
+    body = generate_tweet_body(data)
+    data = adjust_tweet_body(body, data)
+    generate_tweet_body(data)
   end
 
-  def get_episode_number
-    max_length = 10
-    number = @checkin.episode.single? ? '' : @checkin.episode.number
+  def get_episode_number(checkin)
+    checkin.episode.single? ? "" : checkin.episode.number
+  end
 
-    if max_length < number.length
-      number.truncate(max_length)
+  def get_episode_title(checkin)
+    if checkin.episode.single?
+      ""
     else
-      @rest += (max_length - number.length)
-      number
+      checkin.episode.title.presence || ""
     end
   end
 
-  def get_episode_title
-    title = if @checkin.episode.single?
-      ''
-    else
-      @checkin.episode.title.presence || ''
-    end
+  def get_share_hashtag(checkin)
+    checkin.work.twitter_hashtag.presence || ""
+  end
 
-    if @checkin.comment.present?
-      @rest += 10
+  def generate_tweet_body(data)
```

Comment:
Assignment Branch Condition size for generate_tweet_body is too high. [16.16/15]